src/Types/Rational.purs

```diff
@@ -1,11 +1,94 @@
 module Types.Rational
   ( Rational
+  , class RationalComponent
+  , reduce
+  , (%)
+  , recip
+  , numerator
+  , denominator
+  , denominatorAsNat
   ) where
 
+import Prelude
+
+import Data.BigInt (BigInt, fromInt) as BI
```

Comment:
Very minor stylistic thing: we use `BigInt` elsewhere in the codebase and import it qualified as `BigInt`. It would be nice to keep that consistent